builtin/logical/database/path_creds_create.go

```diff
@@ -66,11 +66,15 @@ func (b *databaseBackend) pathCredsCreateRead() framework.OperationFunc {
 			b.Lock()
 			unlockFunc = b.Unlock
 
-			// Create a new DB object
-			db, err = b.createDBObj(ctx, req.Storage, role.DBName)
-			if err != nil {
-				unlockFunc()
-				return nil, fmt.Errorf("cound not retrieve db with name: %s, got error: %s", role.DBName, err)
+			// Check again
+			db, ok = b.getDBObj(role.DBName)
```

Comment:
FWIW, the` createDBObj` function does this check at the beginning of the function, so doesn't necessarily need to happen here.